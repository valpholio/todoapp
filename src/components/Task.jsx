import React from 'react';
import PropTypes from 'prop-types';

/**
 * Representation of the task.
 * @param {string} title - Title of the task.
 * @param {string} text - Text of the task.
 */
const Task = ({title, text}) => {
    return (
        <div>
            <h3 className='task-name'>{title}</h3>
            <div>{text}</div>
        </div>
    );
};

Task.propTypes = {
    title: PropTypes.string,
    text: PropTypes.string,
    isDone: PropTypes.bool,
};

export default Task;
