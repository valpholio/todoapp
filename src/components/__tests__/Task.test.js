import React from 'react';
import renderer from 'react-test-renderer';
import Task from '../Task.jsx';

describe('Task', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<Task title='title' text='text'/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});