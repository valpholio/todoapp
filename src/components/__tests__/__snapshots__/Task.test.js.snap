// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task snapshot renders 1`] = `
<div>
  <h3
    className="task-name"
  >
    title
  </h3>
  <div>
    text
  </div>
</div>
`;
